clarity/contracts/arkadiko-auction-engine-v4-1.clar

```diff
@@ -0,0 +1,440 @@
+      (amounts (unwrap-panic (as-contract (contract-call? vault-manager liquidate vault-id coll-type))))
+
+      (updated-vault (contract-call? .arkadiko-vault-data-v1-1 get-vault-by-id vault-id))
+
+      (auction-id (+ (var-get last-auction-id) u1))
+      (auction {
+        id: auction-id,
+        auction-type: "collateral",
+        vault-id: vault-id,
+        collateral-amount: (get ustx-amount amounts),
+        collateral-token: (get collateral-token updated-vault),
+        collateral-address: (contract-of ft),
+        debt-to-raise: (get vault-debt amounts),
+        discount: (unwrap-panic (contract-call? coll-type get-liquidation-penalty (get collateral-type updated-vault))),
+        total-debt-burned: u0,
+        total-collateral-sold: u0,
+        ended-at: u0,
+      })
+    )
+      (asserts! (>= liquidation-ratio collateral-to-debt-ratio) (err ERR-NO-LIQUIDATION-REQUIRED))
+
+      ;; When freddie liquidates, xSTX tokens are minted in arkadiko-sip10-reserve-v1-1
+      ;; So we need to transfer tokens to the new reserve
+      (if (and (is-eq "STX" (get collateral-token vault)) (> (get stacked-tokens vault) u0))
+        (begin
+          (try! (contract-call? .arkadiko-dao burn-token .xstx-token (get ustx-amount amounts) .arkadiko-sip10-reserve-v1-1))
+          (try! (contract-call? .arkadiko-dao mint-token .xstx-token (get ustx-amount amounts) (contract-of reserve)))
+        )
+        true
+      )
+
+      ;; Add auction
+      (map-set auctions { id: auction-id } auction)
+      (var-set last-auction-id auction-id)
+
+      ;; Try to burn
+      (try! (burn-usda auction-id oracle coll-type vault-manager ft reserve liquidation-pool liquidation-rewards))
+
+      (print { type: "auction", action: "created", data: auction })
+      (ok true)
+    )
+  )
+)
+
+;; @desc burn USDA for vault and get part of collateral
+;; @param auction-id; the auction
+;; @param oracle; oracle
+;; @param coll-type; collateral types
+;; @param vault-manager; vault manager
+;; @param ft; collateral token if not STX vault
+;; @param reserve; reserve to get collateral tokens from
+;; @param liquidation-pool; pool to get USDA from
+;; @param liquidation-rewards; contract to deposit rewards
+(define-public (burn-usda
+  (auction-id uint)
+  (oracle <oracle-trait>)
+  (coll-type <collateral-types-trait>)
+  (vault-manager <vault-manager-trait>)
+  (ft <ft-trait>)
+  (reserve <vault-trait>)
+  (liquidation-pool <liquidation-pool-trait>)
+  (liquidation-rewards <liquidation-rewards-trait>)
+)
+  (let (
+    (auction (get-auction-by-id auction-id))
+    (debt-left (- (get debt-to-raise auction) (get total-debt-burned auction)))
+    (burn-amounts (unwrap-panic (burn-usda-amount auction-id oracle liquidation-pool)))
+    (usda-to-use (get usda-to-use burn-amounts))
+    (collateral-sold (get collateral-to-sell burn-amounts))
+    (collateral-fee (/ (* collateral-sold (var-get auction-fee)) u10000))
```

Comment:
Q: how do we get access to this collateral?

Reply:
Fees will be collected in the contract. Using the 'withdraw-fees' method we can withdraw thes.